samples/bluetooth/hci_uart/src/main.c

```diff
@@ -323,8 +323,8 @@ static int hci_uart_init(const struct device *unused)
 	return 0;
 }
 
-DEVICE_INIT(hci_uart, "hci_uart", &hci_uart_init, NULL, NULL,
-	    APPLICATION, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
+SYS_DEVICE_DEFINE("hci_uart", &hci_uart_init, NULL,
```

Comment:
What is the difference between DEVICE_DEFINE and SYS_DEVICE_DEFINE and why would it be preffered to use the SYS_ versios in this case? 

I'm just curious. Is it something magic with how Z_SYS_NAME?

Reply:
I converted this to SYS_DEVICE_DEFINE as it seemed like the only reason its here is to get `hci_uart_init` called and the lack of any data pointers leaned me in that way.  @tbursztyka can probably give a better understanding of the difference between SYS_DEVICE_DEFINE and DEVICE_DEFINE usage.

Reply:
was wondering about this as well, this should be another candidate for deprecation or renaming, add SYS_ as a prefix does not really say anything about what is actually does.

Reply:
I'll let @tbursztyka comment on that since we added SYS_DEVICE_DEFINE recently:

commit 8d7bb8ffd8faba73628e18e4651e20896da9751e
Author: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>
Date:   Mon Mar 9 11:02:20 2020 +0100

    device: Refactor device structures

Reply:
SYS_DEVICE_DEFINE is for non subsystem devices (spi, i2c, adc, ...), so no data/config nor api, that may expose PM stuff as well.

Maybe we could get rid of it as well, and just use DEVICE_DEFINE. I have no strong opinion on that one.